src/plugin/cesium/cesiumplugin.js

```diff
@@ -94,46 +94,50 @@ plugin.cesium.Plugin.prototype.init = function() {
   plugin.cesium.ionUrl = /** @type {string} */ (os.settings.get(plugin.cesium.SettingsKey.ION_URL,
       plugin.cesium.DEFAULT_ION_URL));
 
-  // set the WebGL renderer to use Cesium
-  os.MapContainer.getInstance().setWebGLRenderer(new plugin.cesium.CesiumRenderer());
-
-  // register the default set of synchronizers
-  var sm = os.webgl.SynchronizerManager.getInstance();
-  sm.registerSynchronizer(os.layer.SynchronizerType.VECTOR, plugin.cesium.sync.VectorSynchronizer);
-  sm.registerSynchronizer(os.layer.SynchronizerType.TILE, plugin.cesium.sync.TileSynchronizer);
-  sm.registerSynchronizer(os.layer.SynchronizerType.IMAGE, plugin.cesium.sync.ImageSynchronizer);
-  sm.registerSynchronizer(os.layer.SynchronizerType.DRAW, plugin.cesium.sync.VectorSynchronizer);
-  sm.registerSynchronizer(os.layer.SynchronizerType.IMAGE_STATIC, plugin.cesium.sync.ImageStaticSynchronizer);
-
-  // add 3D layer group
-  var group = new os.layer.Group();
-  group.setPriority(3);
-  group.setOSType(plugin.cesium.CESIUM_ONLY_LAYER);
-  group.setCheckFunc(function(layer) {
-    if (os.implements(layer, os.layer.ILayer.ID)) {
-      return /** @type {os.layer.ILayer} */ (layer).getOSType() === plugin.cesium.CESIUM_ONLY_LAYER;
-    }
-    return false;
-  });
-
-  os.map.mapContainer.addGroup(group);
-
-  // set up menus
-  plugin.cesium.menu.importSetup();
-
-  // register 3D tiles layers
-  var lcm = os.layer.config.LayerConfigManager.getInstance();
-  lcm.registerLayerConfig(plugin.cesium.tiles.ID, plugin.cesium.tiles.LayerConfig);
-
-  var dm = os.dataManager;
-  dm.registerProviderType(new os.data.ProviderEntry(
-      plugin.cesium.tiles.ID,
-      plugin.cesium.tiles.Provider,
-      plugin.cesium.tiles.TYPE,
-      plugin.cesium.tiles.TYPE));
-  dm.registerDescriptorType(plugin.cesium.tiles.ID, plugin.cesium.tiles.Descriptor);
-
-  var im = os.ui.im.ImportManager.getInstance();
-  im.registerImportDetails(plugin.cesium.tiles.TYPE, true);
-  im.registerImportUI(plugin.cesium.tiles.mime.TYPE, new plugin.cesium.tiles.TilesetImportUI());
+  // check if cesium is the active renderer
+  if (os.settings.get(os.webgl.AbstractWebGLRenderer.ACTIVE_SETTINGS_KEY) == plugin.cesium.Plugin.ID) {
```

Comment:
It might be better to have 3D plugins always register their renderer class with `MapContainer` (just the class, not an instance of it). When `MapContainer` initializes (after plugins have loaded), it will create an instance of the active renderer and call an init function on it. That init function will contain the remaining code in this block. Then plugins don't need to be aware of the internals of how the renderer is selected/activated.

Reply:
Moved the plugin code to the renderer as suggested but didn't do the renderer registration changes yet, still need to test it out first

Reply:
Looks like the MapContainer inits in `onAngularReady_` in `os.ui,map` and I am having trouble hooking this up without regressions within my available time. Let me know if you want me to figure this part out but it may have to wait until next month.